src/ddmd/objc.d

```diff
@@ -117,126 +117,140 @@ struct ObjcSelector
     }
 }
 
-struct Objc_ClassDeclaration
-{
-    // true if this is an Objective-C class/interface
-    bool objc;
+private __gshared Objc _objc;
 
-    // MARK: Objc_ClassDeclaration
-    extern (C++) bool isInterface()
-    {
-        return objc;
-    }
+Objc objc()
+{
+    return _objc;
 }
 
-struct Objc_FuncDeclaration
+interface Objc
 {
-    FuncDeclaration fdecl;
-    // Objective-C method selector (member function only)
-    ObjcSelector* selector;
-
-    extern (D) this(FuncDeclaration fdecl)
+    static void _init()
```

Comment:
Oh! I apologise as I forgot, this static method should definitely be `extern(C++) ` at the very least, because gdc's entrypoint is in C++.